geometry/rotation.hpp

```diff
@@ -27,10 +27,32 @@ template<typename FromFrame, typename ToFrame>
 class Rotation : public LinearMap<FromFrame, ToFrame> {
  public:
   explicit Rotation(Quaternion const& quaternion);
-  explicit Rotation(R3x3Matrix const& matrix);
   template<typename Scalar>
   Rotation(quantities::Angle const& angle,
            Bivector<Scalar, FromFrame> const& axis);
+
+  // Construct a rotation from the axes of |ToFrame|, expressed in |FromFrame|.
+  template<int rank_x, int rank_y, int rank_z,
+           typename = std::enable_if_t<
+                          (rank_x + rank_y + rank_z) % 2 == 0 &&
+                          !std::is_same<FromFrame, ToFrame>::value>>
```

Comment:
Let's do the rank with a `static_assert` in the body and the frame check using additional template parameters.